homeassistant/components/purpleair/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The PurpleAir integration."""
+from __future__ import annotations
+
+from aiopurpleair.models.sensors import SensorModel
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import PurpleAirDataUpdateCoordinator
+
+MAP_URL_BASE = "https://map.purpleair.com/1/mAQI/a10/p604800/cC0"
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up PurpleAir from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    coordinator = PurpleAirDataUpdateCoordinator(hass, entry)
+    await coordinator.async_config_entry_first_refresh()
+    hass.data[DOMAIN][entry.entry_id] = coordinator
```

Comment:
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```

Its nicer to avoid adding anything to `hass.data` until the setup is successful.